src/Microsoft.DotNet.CodeFormatting.Tests/FormatOptionsParsingTests.cs

```diff
@@ -0,0 +1,152 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.IO;
+using System.Linq;
+using System.Collections.Generic;
+
+using CodeFormatter;
+
+using CommandLine;
+
+using Xunit;
+
+namespace Microsoft.DotNet.CodeFormatting.Tests
+{
+    public class FormatOptionsParsingTests
+    {
+        private FormatOptions Parse(out int result, params string[] args)
+        {
+            result = 1;
+            FormatOptions options = null;
+
+            using (var writer = new StringWriter())
+            {
+                var parser = new Parser(settings =>
+                {
+                    // CommandLine library help output is not thread-safe. We
+                    // disable console output entirely by providing a null writer
+                    settings.HelpWriter = null;
+                });
+
+                result = parser.ParseArguments<FormatOptions>(args)
+                    .Return(
+                    (FormatOptions parsedOptions) => { options = parsedOptions; return 0; },
+                    errs => ReportErrors(errs));
+            }
+
+            return options;
+        }
+
+        private int ReportErrors(IEnumerable<Error> errs)
+        {
+            return 1;
+        }
+
+        [Fact]
+        public void EnableRule()
+        {
+            int result;
+
+            // CodeFormatter format --enable enabledRule -t test.csproj
+            var options = Parse(out result, "format", "--enable", "enabledRule", "-t", "test.csproj");
+
+            Assert.Equal(0, result);
+            Assert.NotNull(options);
+            Assert.True(options.RuleMap["enabledRule"]);
+            Assert.True(options.RuleMap.Count == 1);
```

Comment:
I think equalities are more clearly asserted with `Assert.Equal(1, options.RuleMap.Count)` (here and below)